App.js

```diff
@@ -1,11 +1,67 @@
-import React from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import React, { Component } from 'react';
+import { StyleSheet, View, KeyboardAvoidingView, Alert, FlatList } from 'react-native';
+import AddTaskMenu from 'ToDoProject/Components/AddTaskMenu'
+import TaskList from 'ToDoProject/Components/TaskList';
+import { Header } from 'react-native-elements';
+import Task from 'ToDoProject/Components/Task';
+import AppearingTextInput from 'ToDoProject/Components/AppearingTextInput';
+
+export default class App extends Component {
+  constructor(props) {
+    super(props);
+    this.id = 0;
+    this.taskList = [];
+    this.state = {
+      tasks: this.taskList,
+      isInputVisible: false,
+    }
+  }
+
+  addTask = (task) => {
+    if (task != '') {
+      this.taskList.push({
+        key: this.id.toString(),
+        description: task,
+        isDone: false
+      });
+      this.setState({
+        tasks: this.taskList,
+      });
+      this.id++;
+    }
+  }
+
+  onDelete = (taskNumber) => {
+    for (var i = 0; i < this.taskList.length; ++i) {
```

Comment:
easier will be 
let tasks = this.state.tasklist.filter(t=>t.key!=taskNumber)
this.setState({tasks});